### ISSUE TITLE
lstat /host_mnt: function not implemented when using :z

### ISSUE THREAD

**Issue by billyfoss** [docker/for-win#6634]
Upgraded to 2.3.0.2 today and found that mounts are not working as they did before.  

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: B86D9DE7-77D8-4B7E-A1FF-1228340E07E0/20200511214828

### Expected behavior
This output was from without :z
```
$ docker run --rm -it -v C:/tmp:/tmp:z centos:7 /bin/bash
[root@b9a769b285d4 /]# ls -alZ /tmp
drwxrwxrwx root root ?                                .
drwxr-xr-x root root ?                                ..
```

### Actual behavior

```
$ docker run --rm -it -v C:/tmp:/tmp:z centos:7 /bin/bash
docker: Error response from daemon: error evaluating symlinks from mount source "/host_mnt/c/tmp": lstat /host_mnt: function not implemented.
```

### Information
This is happening consistently now.  Previously, I had tried Edge and had occasional function not implemented error messages.  Restarting Docker resolved the issue before, but this appears to be consistent with the new version of the Stable channel.
I am running within Ubuntu 18.04 using WSL1

  - Windows Version: Windows 10 Pro (1903 - 18362.778)
  - Docker Desktop Version: 2.3.0.2
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior

  1. Install new version
  2. Run with :z
```
docker run --rm -it -v C:/tmp:/tmp:z centos:7 /bin/bash
```

**Comment by djs55** [comment]
Thanks for your report. Could I just check: does it work ok for you without `:z`? The Linux VM inside Docker Desktop does not enable SELinux so labelling ought not to be necessary.

**Comment by Gemorroj** [comment]
link to version 2.2.0.5
https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2205

**Comment by dquintela** [comment]
I have the same issue with the docker for windows update I did yesterday.

```
C:\data\projects\audit-service\service>docker run -v c:\data\projects\audit-service\service\target\audit-service-1.0.0-SNAPSHOT-native-image-source-jar:/project:z --env LANG=C --rm quay.io/quarkus/ubi-quarkus-native-image:20.0.0-java11 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=Cp1252 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:+JNI -jar audit-service-1.0.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http -H:-UseServiceLoaderFeature -H:+StackTrace audit-service-1.0.0-SNAPSHOT-runner
docker: Error response from daemon: error evaluating symlinks from mount source "/host_mnt/c/data/projects/audit-service/service/target/audit-service-1.0.0-SNAPSHOT-native-image-source-jar": lstat /host_mnt: function not implemented.
```

Removed the ":v" and with works
```
C:\data\projects\audit-service\service>docker run -v c:\data\projects\audit-service\service\target\audit-service-1.0.0-SNAPSHOT-native-image-source-jar:/project --env LANG=C --rm quay.io/quarkus/ubi-quarkus-native-image:20.0.0-java11 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Dfile.encoding=Cp1252 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:+JNI -jar audit-service-1.0.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http -H:-UseServiceLoaderFeature -H:+StackTrace audit-service-1.0.0-SNAPSHOT-runner
Build on Server(pid: 27, port: 34685)*
[audit-service-1.0.0-SNAPSHOT-runner:27]    classlist:  10,487.90 ms,  1.00 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]        (cap):   1,042.24 ms,  1.00 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]        setup:   3,066.04 ms,  1.00 GB
10:09:09,846 INFO  [org.jbo.res.res.i18n] RESTEASY002225: Deploying javax.ws.rs.core.Application: class com.truphone.rspplatform.audit.service.resources.AuditInvocationApplication
10:09:31,877 INFO  [org.jbo.threads] JBoss Threads version 3.1.1.Final
[audit-service-1.0.0-SNAPSHOT-runner:27]   (typeflow):  40,911.71 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]    (objects):  36,399.59 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]   (features):     749.38 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]     analysis:  81,218.45 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]     (clinit):   1,351.74 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]     universe:   5,070.65 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]      (parse):  11,913.59 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]     (inline):  10,291.82 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]    (compile):  56,918.56 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]      compile:  81,995.23 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]        image:   4,103.80 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]        write:   3,226.69 ms,  1.54 GB
[audit-service-1.0.0-SNAPSHOT-runner:27]      [total]: 189,638.42 ms,  1.54 GB

```

**Comment by Tieantono** [comment]
Not only Z drive I guess, it also happened on my C or D drives.
I've got a similar problem when mounting a volume on the same Docker version `2.3.0.2` when I want to run a `docker-compose` file from [here](https://github.com/jitsi/docker-jitsi-meet#quick-start)

Error messages when running `docker-compose up -d` command:
```
ERROR: for docker-jitsi-meet_prosody_1  Cannot start service prosody: error evaluating symlinks from mount source "/host_mnt/d/Jitsi/.jitsi-meet-cfg/prosody/config": lstat /host_mnt: function not implemented

ERROR: for docker-jitsi-meet_web_1  Cannot start service web: error evaluating symlinks from mount source "/host_mnt/d/Jitsi/.jitsi-meet-cfg/transcripts": lstat /host_mnt: function not implemented

ERROR: for prosody  Cannot start service prosody: error evaluating symlinks from mount source "/host_mnt/d/Jitsi/.jitsi-meet-cfg/prosody/config": lstat /host_mnt: function not implemented

ERROR: for web  Cannot start service web: error evaluating symlinks from mount source "/host_mnt/d/Jitsi/.jitsi-meet-cfg/transcripts": lstat /host_mnt: function not implemented
```

**Comment by dquintela** [comment]
Just downgraded back to Docker Desktop Community 2.2.0.5 and it's working again.
https://docs.docker.com/docker-for-windows/release-notes/

**Comment by Tieantono** [comment]
> Just downgraded back to Docker Desktop Community 2.2.0.5 and it's working again.
> https://docs.docker.com/docker-for-windows/release-notes/

Yes, it is working on version 2.2.05. Thank you.

**Comment by rincler** [comment]
The same problem on 2.3.0.2

**Comment by stephen-turner** [comment]
Thanks for the reports. @thaJeztah tells us that this is due to a known bug in the upstream engine: https://github.com/moby/moby/issues/40944. We'll be able to fix it when the change has made it into an engine patch release.

**Comment by yehoshuadimarsky** [comment]
> The same problem on 2.3.0.2

same here

**Comment by thaJeztah** [comment]
If you're having this issue with the `:z` or `:Z` options (SELinux labeling), the workaround is to remove that option; even with the issue fixed, those options won't have an effect when using docker desktop (the fix linked above (to be released in the next engine patch release) is to ignore the error of the underlying filesystem doesn't support xattrs / SELinux labels which is the case with docker desktop)

**Comment by jackdcasey** [comment]
To anyone running into this issue when using the serverless framework, I was able to get things functional by downgrading to version [2.2.0.5](https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2205).

Hope this prevents a headache for anyone else running into this!  😁

**Comment by CaliforniaMountainSnake** [comment]
I have the same problem.
`
Cannot start service mysql: error evaluating symlinks from mount source "/host
_mnt/d/Servers/www/project/logs/mysql": lstat /host_mnt: function not implemented
`

**Comment by thaJeztah** [comment]
> Yes, it is working on version 2.2.0.5. Thank you.

Actually; @stephen-turner double checking w.r.t. the SELinux (`:z` / `:Z`); the regression was in Docker 19.03.7 and 19.03.8 (https://github.com/moby/moby/pull/40547). Looking at the [release notes](https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2204), Docker Desktop 2.2.0.5 already had engine 19.03.8, so if it works with 2.2.0.5, but doesn't work with 2.3.0.x, I think it's a different issue

**Comment by valentijnscholten** [comment]
It also fails when using docker-compose, without any `:z` in mounts specified.

**Comment by mat007** [comment]
Thanks for the reports, we understand where this is coming from now and are working on a fix!

**Comment by robrechta** [comment]
any idea when the fix of this issue is planned? I see a build 2.3.0.3 appearing, but this ticket is still open... So I don't want to loose time of doing an upgrade, and then back a downgrade to 2.2.0.5...

**Comment by mat007** [comment]
The fix has now been released in Docker Desktop Stable 2.3.0.3, I’ll go ahead and close this issue.
